pkg/reconciler/knativeeventing/common/replicasenvvarstransform.go

```diff
@@ -53,13 +62,30 @@ func ReplicasEnvVarsTransform(client unstructuredGetter) mf.Transformer {
 			// Keep the existing number of replicas in the cluster for the deployment
 			apply.Spec.Replicas = current.Spec.Replicas
 
+			// Preserve the env vars in the existing cluster
 			for index := range current.Spec.Template.Spec.Containers {
 				currentContainer := current.Spec.Template.Spec.Containers[index]
-				for j := range apply.Spec.Template.Spec.Containers {
-					applyContainer := &apply.Spec.Template.Spec.Containers[j]
-					if currentContainer.Name == applyContainer.Name {
-						applyContainer.Env = currentContainer.Env
+				found, containerIndex := nameExistsContainers(currentContainer.Name, apply.Spec.Template.Spec.Containers)
+				if found {
```

Comment:
You can spare yourself a level of indentation if you do 

```golang
if !found {
  continue
}

applyContainer := ...
```

Reply:
As I re-implement `findContainer(name string, containers []corev1.Container) *corev1.Container`, we can determine the container is found or not, by checking if the returned container is nil or not.